### PR TITLE
DPL-937 - Updates to RVI BCL pipeline

### PR DESCRIPTION
**Pull request by BenTopping**
#### Changes proposed in this pull request

- Adds RVI prefix to bait capture library pipeline
- Updates robot verifications for rvi bcl pipeline
- Adds custom presenter to RVI cDNA plate to allow custom 'Manual transfer' state change.

#### Instructions for Reviewers

_[All PRs] - Confirm PR template filled_  
_[Feature Branches] - Review code_  
_[Production Merges to `main`]_  
 &nbsp; &nbsp; \- _Check story numbers included_  
 &nbsp; &nbsp; \- _Check for debug code_  
 &nbsp; &nbsp; \- _Check version_

### CODE REVIEW

File: app/models/presenters/rvi_cdna_xp_presenter.rb [hunk 1]
```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+module Presenters
+  # Used for RVI cDNA XP plates. Used for a custom 'Manual Transfer' state transition.
+  # N.B. 'Manual Transfer' is used for the RVI BCL pipeline for the cDNA XP Plate because the plates
+  # are inaccessible in the bed at the time of transfer so cannot be transferred via
+  # bed verifications. Also we cannot set its state to 'Passed' on initial bed verification
+  # because there are steps on other plates that need to be completed before the cDNA XP Plate can be transferred.
+  # and we do not want to misreport the plate being passed if an issue arises with the steps before hand.
+  class RviCdnaXpPresenter < PlatePresenter
```
Comment (andrewsparkes) [inline app/models/presenters/rvi_cdna_xp_presenter.rb:10]: Minor thing but the capitalisation in the classname looks odd to me.
`RVIcDNAXpPresenter`?
RVI and DNA are already initialisms of longer titles.
c in cDNA means complimentary and is written in lower case.
Dunno if doing that causes any class loading issues mind.

File: app/models/presenters/rvi_cdna_xp_presenter.rb [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+      # Statemachines ends up throwing a warning, which while we can disable, we're probably
+      # safer avoiding the conflict, especially as we don't actually call these methods directly
+      # anyway.
+      event :fail_labware, human_name: 'Fail' do
+        transition %i[pending started passed] => :failed
+      end
+
+      # These are the states, which are really the only things we need ...
+      state :pending do
+        include Statemachine::StateAllowsChildCreation
```
Comment (andrewsparkes) [inline app/models/presenters/rvi_cdna_xp_presenter.rb:38]: Just checking, at the point they want to click on the 'Add Ligand' plate button is the cDNA plate still in pending state (or in started or another later state)? I'd delay allowing ChildCreation as long as possible. But if it is in pending state this looks ok.

File: app/models/presenters/rvi_cdna_xp_presenter.rb [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  # bed verifications. Also we cannot set its state to 'Passed' on initial bed verification
+  # because there are steps on other plates that need to be completed before the cDNA XP Plate can be transferred.
+  # and we do not want to misreport the plate being passed if an issue arises with the steps before hand.
+  class RviCdnaXpPresenter < PlatePresenter
+    include Statemachine::Shared
+
+    # Same state machine as permissive plate presenter, but with a custom 'Manual Transfer' state transition.
+    state_machine :state, initial: :pending do
+      event :take_default_path, human_name: 'Manual Transfer' do
+        transition pending: :started
```
Comment (andrewsparkes) [inline app/models/presenters/rvi_cdna_xp_presenter.rb:16]: should there be 2 transitions here? if this presenter is for one specific plate shouldn't there by only one state change by manual transfer? or is the first backup for if the robot (bed verification) is not available for the first state transition?

**Review by andrewsparkes — APPROVED** [review]

File: app/models/presenters/rvi_cdna_xp_presenter.rb (hunk 1 above)
Comment (BenTopping) [reply app/models/presenters/rvi_cdna_xp_presenter.rb:10]: I agree, not the nicest capitalisation, but having the name as anything else causes loading issues unfortunately.

File: app/models/presenters/rvi_cdna_xp_presenter.rb (hunk 2 above)
Comment (BenTopping) [reply app/models/presenters/rvi_cdna_xp_presenter.rb:38]: Yes good spot, will change so only 'started' and 'passed' can have child creation.

File: app/models/presenters/rvi_cdna_xp_presenter.rb (hunk 3 above)
Comment (BenTopping) [reply app/models/presenters/rvi_cdna_xp_presenter.rb:16]: Yeah i included started: :passed in the case the bed verification cant be completed. But that should never happen so probably safe to remove?

**Pull request merged**